[69] 프론트 에러핸들링

soozynn opened the pull request:
# [69] [Task] 프론트 에러 핸들링

## 노션 칸반 링크
- [[Task] 프론트 에러 핸들링](https://www.notion.so/vanillacoding/Task-41b650dff7914291ba467ec315d685aa)

## 카드에서 구현 혹은 해결하려는 내용
- Feat: import 정렬
- Feat: action에 대한 성공/실패 모달 구현

## 테스트 방법
- action 시 상황에 따라 error / success 모달이 오픈되는지 확인

## 기타 사항
- dev를 최신화하고 확인도 여러 번 했는데 혹시 제가 놓친 부분이 있을 수도 있을 것 같아서 한번 더 체크해주시면 감사합니다!

lazymole commented on package.json:
```diff
@@ -70,5 +70,6 @@
     "prettier": "^2.5.1"
-  }
+  },
+  "proxy": "http://localhost:8000"
```
```suggestion
```

lazymole commented on src/components/Chat/MessageBox.js:
```diff
@@ -68,9 +68,11 @@ const DateWrapper = styled.div`
   const { profile, displayName, message, createdAt } = data;
-
+  const [myChat, setMyChat] = useState();
+  // 로그인한 유저는 텍스트랑 날짜만
```
```suggestion
```

lazymole commented on src/components/Chat/MessageBox.js:
```diff
@@ -68,9 +68,11 @@ const DateWrapper = styled.div`
+  // 로그인한 유저는 텍스트랑 날짜만
   return (
     <Container>
+      {myChat}
```
이 부분이 혼자 동떨어져 나와있는 게 이상하네요. 다른 태그로 감싸주면 좋겠습니다. setMyChat이 안 쓰이고 있던데 나중에 추가 기능을 제작하기 위한 부분이지요?

Commit:
[69] 프론트 에러핸들링

- Feat: dev 최신화 후 다시 업데이트

[69] 프론트 에러핸들링

- Feat: app 에러핸들링 추가

[69] 프론트 에러핸들링

- Feat: css 속성 수정

lazymole commented on src/components/Mandal/MandalPage.js:
```diff
@@ -86,7 +92,10 @@ export default function MandalPage() {
+  const goalListErrorState = useSelector(state => state.goalList.error);
+  const mandalErrorState = useSelector(state => state.mandal.error);
+  const todoErrorState = useSelector(state => state.todo.error);
   useEffect(() => {
```
```suggestion

  useEffect(() => {
```

lazymole commented on src/components/Mandal/view/SubMandal.js:
```diff
@@ -4,9 +4,9 @@ import styled from "styled-components";
-import { showBoxes } from "./utils";
 import { VIEW_OPTION } from "../../../constants";
-import Modal from "../../Modal";
+import Modal from "../../Modal/Modal";
+import { showBoxes } from "./utils";
 import Todo from "./Todo";
```
컴포넌트끼리 묶어주면 더 좋을 것 같은데, 지금 자동 포매팅이 디렉토리 기반으로 되어있나보네요.

lazymole commented on src/components/Modal/ErrorModal.js:
```diff
@@ -0,0 +1,97 @@
+    border-radius: 6px;
+  }
+
+  div {
+    div,
+    p {
+      margin: 4px;
+    }
+  }
```
자식 div를 지정해주기 위함일까요? 자식만 고르고 싶을 때는 `>`를 셀렉터로  이용할 수 있어요.
아래와 같은 코드 예시에서 `.something > input` 을 셀렉터로 주면 `<input type="submit">`만 선택됩니다.

```
<div class="something">
  <div>
    <input type="text" />
  </div>
  <input type="submit" />
</div>
```

lazymole commented on src/components/Modal/ErrorModal.js:
```diff
@@ -0,0 +1,97 @@
+    background: ${props => props.background || "#e85e6c"};
+  }
+
+  div div {
+    margin-top: 15px;
+    color: #a6aab6;
+  }
```
div가 두 개 있어서 이상해요! 위의 코멘트 참고해주시면 좋겠습니다.
